pkg/oom/v1/v1.go

```diff
@@ -25,16 +25,14 @@ import (
 
 	"github.com/containerd/cgroups/v3/cgroup1"
 	eventstypes "github.com/containerd/containerd/v2/api/events"
-	"github.com/containerd/containerd/v2/core/runtime"
-	"github.com/containerd/containerd/v2/core/runtime/v2/shim"
 	"github.com/containerd/containerd/v2/pkg/oom"
 	"github.com/containerd/log"
 	"golang.org/x/sys/unix"
 )
 
 // New returns an epoll implementation that listens to OOM events
 // from a container's cgroups.
-func New(publisher shim.Publisher) (oom.Watcher, error) {
+func New(publisher chan<- interface{}) (oom.Watcher, error) {
```

Comment:
what about using `events.Event` instead of `interface` here?

Reply:
Can you clarify this?

`oom` package writes `eventstypes.TaskOOM` to this channel.
But the channel and passed from the shim directly - https://github.com/containerd/containerd/blob/1641c7571f0b60edbf9de69b7fc824911e89ca64/cmd/containerd-shim-runc-v2/task/service.go#L77
The channel is generic because it handles different event types within shims.

Having typed channel in `oom` package would require us casting it from `TaskOOM` to `interface{}` somewhere.

Reply:
> The channel is generic because it handles different event types within shims.

OK. I was thinking that `events.Event` is more specific type here, even if it's alias of `interface{}`.
It's just for review or maintain purpose.